uvicorn/supervisors/multiprocess.py

```diff
@@ -39,9 +40,14 @@ def signal_handler(self, sig, frame):
 
     def run(self):
         self.startup()
-        self.should_exit.wait()
+        while not self.should_exit.wait(0.25):
+            if not self.workers_alive():
+                self.should_exit.set()
         self.shutdown()
 
+    def workers_alive(self):
```

Comment:
Is it possible for this function to return negative number, eg. when old processes haven't finished shutdown but new ones have already spawned?

Reply:
I dont see a situation where it could happen, the max number of workers is the one when it's initialized and set to `self.workers_number`, after that it can only decrease, 

for this number to be negative there would be the need for new workers to be spawned, but in our case it only happens once at start, if it dies it dies